packages/grid/x-data-grid/src/hooks/features/rowSelection/utils.ts

```diff
@@ -0,0 +1,15 @@
+import type { DataGridProcessedProps } from '../../../models/props/DataGridProps';
+import { GridSignature } from '../../utils/useGridApiEventHandler';
+
+export function isMultipleRowSelectionEnabled(
+  props: Pick<
+    DataGridProcessedProps,
+    'signature' | 'disableMultipleRowSelection' | 'checkboxSelection'
+  >,
+) {
+  if (props.signature === GridSignature.DataGrid) {
+    // DataGrid Community has multiple row selection enabled only if checkbox selection is enabled.
+    return props.checkboxSelection && props.disableMultipleRowSelection !== true;
+  }
+  return !props.disableMultipleRowSelection;
+}
```

Comment:
This is a bit messy, but the main change in this PR is that you can now pass the `disableMultipleRowSelection` prop to MIT `DataGrid`.
The default behavior hasn't changed, so this PR does not bring any breaking changes.